claude-code: build failure as of `8ddddcd3`

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

```
nixpkgs on  master on ☁️  skainsworth@gmail.com took 9s
❯ nix-build -A claude-code
this derivation will be built:
  /nix/store/58anmddnv7b1yznhdwkjjvxxqqck8cb5-claude-code-0.2.94.drv
building '/nix/store/58anmddnv7b1yznhdwkjjvxxqqck8cb5-claude-code-0.2.94.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/m9d1kjlbkicgg3gk3jh5164yq65p5gi7-source
source root is source
Running phase: patchPhase
Executing npmConfigHook
Configuring npm
Validating consistency between /private/tmp/nix-build-claude-code-0.2.94.drv-0/source/package-lock.json and /nix/store/75h40mdcsxgzvcm28bx740zz6y5yqs4j-claude-code-0.2.94-npm-deps/package-lock.json
Setting npm_config_cache to /nix/store/75h40mdcsxgzvcm28bx740zz6y5yqs4j-claude-code-0.2.94-npm-deps
Installing dependencies

added 40 packages, and audited 41 packages in 366ms

11 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
patching script interpreter paths in node_modules
node_modules/prebuild-install/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/bin/node"
node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/bin/node"
node_modules/better-sqlite3/deps/download.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/xhcgnphdwfg81j79nhspm0876cxglyj3-bash-5.2p37/bin/bash"
node_modules/rc/cli.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/bin/node"
npm error code 1
npm error path /private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3
npm error command failed
npm error command sh -c prebuild-install || node-gyp rebuild --release
npm error make: Entering directory '/private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3/build'
npm error   TOUCH ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error   ACTION deps_sqlite3_gyp_locate_sqlite3_target_copy_builtin_sqlite3 ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error   TOUCH Release/obj.target/deps/locate_sqlite3.stamp
npm error   CC(target) Release/obj.target/sqlite3/gen/sqlite3/sqlite3.o
npm error   LIBTOOL-STATIC Release/sqlite3.a
npm error rm ba23eeee118cd63e16015df367567cb043fed872.intermediate
npm error make: Leaving directory '/private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3/build'
npm error prebuild-install warn install EACCES: permission denied, access '/nix/store/75h40mdcsxgzvcm28bx740zz6y5yqs4j-claude-code-0.2.94-npm-deps'
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@11.0.0
npm error gyp info using node@22.14.0 | darwin | arm64
npm error gyp info find Python using Python version 3.12.9 found at "/nix/store/xzjghvsg4fhr2vv6h4scihsdrgk4i76w-python3-3.12.9/bin/python3"
npm error gyp WARN read config.gypi ENOENT: no such file or directory, open '/nix/store/hhq2ccjzbffxsfzjzn7bcix41h985xf6-nodejs-22.14.0-source/include/node/config.gypi'
npm error gyp info spawn /nix/store/xzjghvsg4fhr2vv6h4scihsdrgk4i76w-python3-3.12.9/bin/python3
npm error gyp info spawn args [
npm error gyp info spawn args '/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm error gyp info spawn args 'binding.gyp',
npm error gyp info spawn args '-f',
npm error gyp info spawn args 'make',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3/build/config.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm error gyp info spawn args '-I',
npm error gyp info spawn args '/nix/store/hhq2ccjzbffxsfzjzn7bcix41h985xf6-nodejs-22.14.0-source/common.gypi',
npm error gyp info spawn args '-Dlibrary=shared_library',
npm error gyp info spawn args '-Dvisibility=default',
npm error gyp info spawn args '-Dnode_root_dir=/nix/store/hhq2ccjzbffxsfzjzn7bcix41h985xf6-nodejs-22.14.0-source',
npm error gyp info spawn args '-Dnode_gyp_dir=/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp',
npm error gyp info spawn args '-Dnode_lib_file=/nix/store/hhq2ccjzbffxsfzjzn7bcix41h985xf6-nodejs-22.14.0-source/$(Configuration)/node.lib',
npm error gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3',
npm error gyp info spawn args '-Dnode_engine=v8',
npm error gyp info spawn args '--depth=.',
npm error gyp info spawn args '--no-parallel',
npm error gyp info spawn args '--generator-output',
npm error gyp info spawn args 'build',
npm error gyp info spawn args '-Goutput_dir=.'
npm error gyp info spawn args ]
npm error gyp info spawn make
npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm error sh: ./gyp-mac-tool: /usr/bin/env: bad interpreter: Operation not permitted
npm error make: *** [deps/sqlite3.target.mk:244: Release/sqlite3.a] Error 126
npm error gyp ERR! build error
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:216:23)
npm error gyp ERR! System Darwin 24.4.0
npm error gyp ERR! command "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/bin/node" "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
npm error gyp ERR! cwd /private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3
npm error gyp ERR! node -v v22.14.0
npm error gyp ERR! node-gyp -v v11.0.0
npm error gyp ERR! not ok
npm error Log files were not written due to an error writing to the directory: /nix/store/75h40mdcsxgzvcm28bx740zz6y5yqs4j-claude-code-0.2.94-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
error: builder for '/nix/store/58anmddnv7b1yznhdwkjjvxxqqck8cb5-claude-code-0.2.94.drv' failed with exit code 1;
       last 25 log lines:
       > npm error gyp info spawn args '-Dnode_gyp_dir=/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp',
       > npm error gyp info spawn args '-Dnode_lib_file=/nix/store/hhq2ccjzbffxsfzjzn7bcix41h985xf6-nodejs-22.14.0-source/$(Configuration)/node.lib',
       > npm error gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3',
       > npm error gyp info spawn args '-Dnode_engine=v8',
       > npm error gyp info spawn args '--depth=.',
       > npm error gyp info spawn args '--no-parallel',
       > npm error gyp info spawn args '--generator-output',
       > npm error gyp info spawn args 'build',
       > npm error gyp info spawn args '-Goutput_dir=.'
       > npm error gyp info spawn args ]
       > npm error gyp info spawn make
       > npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
       > npm error sh: ./gyp-mac-tool: /usr/bin/env: bad interpreter: Operation not permitted
       > npm error make: *** [deps/sqlite3.target.mk:244: Release/sqlite3.a] Error 126
       > npm error gyp ERR! build error
       > npm error gyp ERR! stack Error: `make` failed with exit code: 2
       > npm error gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:216:23)
       > npm error gyp ERR! System Darwin 24.4.0
       > npm error gyp ERR! command "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/bin/node" "/nix/store/2ribxb3gi87gj4331m6k0ydn0z90zfi7-nodejs-22.14.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
       > npm error gyp ERR! cwd /private/tmp/nix-build-claude-code-0.2.94.drv-0/source/node_modules/better-sqlite3
       > npm error gyp ERR! node -v v22.14.0
       > npm error gyp ERR! node-gyp -v v11.0.0
       > npm error gyp ERR! not ok
       > npm error Log files were not written due to an error writing to the directory: /nix/store/75h40mdcsxgzvcm28bx740zz6y5yqs4j-claude-code-0.2.94-npm-deps/_logs
       > npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
       For full logs, run 'nix-store -l /nix/store/58anmddnv7b1yznhdwkjjvxxqqck8cb5-claude-code-0.2.94.drv'.
```

### Steps to reproduce

checkout 8ddddcd3c03d276e8166faa7aabb66aea0ecf25f, `nix-build -A claude-code`

### Expected behaviour

the build to succeed

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.4.0, macOS 15.4.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(samuelainsworth): `"darwin, home-manager, nixpkgs-23.11pre511546.844ffa82bbe2"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/samuelainsworth/.nix-defexpr/channels/nixpkgs`

### Notify maintainers

@malob 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/cl/claude-code/package.nix
+++ b/pkgs/by-name/cl/claude-code/package.nix
@@ -2,11 +2,14 @@
   lib,
   buildNpmPackage,
   fetchzip,
+  nodejs_20,
 }:
 
 buildNpmPackage rec {
   pname = "claude-code";
   version = "0.2.94";
+
+  nodejs = nodejs_20;
 
   src = fetchzip {
     url = "https://registry.npmjs.org/@anthropic-ai/claude-code/-/claude-code-${version}.tgz";

